pandas/core/window/rolling.py

```diff
@@ -410,18 +392,44 @@ def _insert_on_column(self, result: "DataFrame", obj: "DataFrame"):
                 # insert at the end
                 result[name] = extra_col
 
-    def _center_window(self, result: np.ndarray, window) -> np.ndarray:
+    def calculate_center_offset(self, window, center: bool) -> int:
```

Comment:
pls type window: Union[np.ndarray, int]

Reply:
Doing that causes an error in typing validation:
```
pandas/core/window/rolling.py:417: error: Argument 1 to "len" has incompatible type "Union[Any, int]"; expected "Sized"  [arg-type]
pandas/core/window/rolling.py:2300: error: Argument 1 to "len" has incompatible type "Union[Any, int]"; expected "Sized"  [arg-type]
```